rir/src/compiler/analysis/verifier.cpp

```diff
@@ -187,18 +187,7 @@ class TheVerifier {
         if (auto phi = Phi::Cast(i)) {
             phi->eachArg([&](BB* input, Value* v) {
                 if (auto iv = Instruction::Cast(v)) {
-                    if (iv == phi) {
-                        // Note: can happen in a one-block loop, but only if it
-                        // is not edge-split
-                        std::cerr << "Error at instruction '";
-                        i->print(std::cerr);
-                        std::cerr << "': input '";
-                        iv->printRef(std::cerr);
-                        std::cerr << "' phi has itself as input\n";
-                        ok = false;
-                    }
-
-                    if (input == phi->bb()) {
+                    if (input == phi->bb() && iv != phi) {
```

Comment:
while I agree that a phi can be it's own input, a phi can never have it's own block as input. this would violate edge-split. so this should just be `if (input == phi->bb())`.

Reply:
You're right, I got confused with the "input BB" vs. the BB where the input value is actually located